src/d/actor/d_a_npc_roten.cpp

```diff
@@ -5,322 +5,2883 @@
+    mpMorf->setAnm(pAnmRes, param_2, param_3, 1.0f, 0.0f, -1.0f, NULL);
+    field_0x9C0 = param_1;
+    field_0x998 = param_2;
 }
 
 /* 000039C8-00003A80       .text setAnmTbl__12daNpcRoten_cFP12sRotenAnmDat */
-void daNpcRoten_c::setAnmTbl(sRotenAnmDat*) {
-    /* Nonmatching */
+bool daNpcRoten_c::setAnmTbl(sRotenAnmDat* param_1) {
+    field_0x9C1  &= ~0x1;
+    if(param_1->field_0x00 == 0xFF) {
+        field_0x988 = 0;
+        return true;
+    }
+
+    field_0x988 = param_1;
+    field_0x9C2 = field_0x988->field_0x02;
+    int temp = 2;
+    if(field_0x9C2 > 0) {
+        temp = 0;
+    }
+    if(field_0x9C0 != field_0x988->field_0x00 || temp == 0 || field_0x998 == 0) {
+        setAnm(field_0x988->field_0x00, temp, field_0x988->field_0x01);
+    }
+
+    return false;
 }
 
 /* 00003A80-00003AA4       .text isHaitatuItem__12daNpcRoten_cFUc */
-void daNpcRoten_c::isHaitatuItem(unsigned char) {
-    /* Nonmatching */
+BOOL daNpcRoten_c::isHaitatuItem(u8 itemNo) {
+    return FLOWER_1 <= itemNo && itemNo <= XXX_039;
 }
 
 /* 00003AA4-00003AC8       .text isKoukanItem__12daNpcRoten_cFUc */
-void daNpcRoten_c::isKoukanItem(unsigned char) {
-    /* Nonmatching */
+BOOL daNpcRoten_c::isKoukanItem(u8 itemNo) {
+    return FLOWER_1 <= itemNo && itemNo <= PRESIDENT_STATUE;
 }
 
 /* 00003AC8-00003B30       .text isGetMap__12daNpcRoten_cFUc */
-void daNpcRoten_c::isGetMap(unsigned char) {
-    /* Nonmatching */
+BOOL daNpcRoten_c::isGetMap(u8 itemNo) {
+    if(mNpcNo == 1 && !dComIfGs_isEventBit(0x3E04) && itemNo == PRESIDENT_STATUE) {
+        return true;
+    }
+
+    return false;
 }
 
 /* 00003B30-00003C7C       .text XyEventCB__12daNpcRoten_cFi */
-void daNpcRoten_c::XyEventCB(int) {
-    /* Nonmatching */
+s16 daNpcRoten_c::XyEventCB(int equippedItemIdx) {
+    s16 eventIdx;
+
+    int itemNo = dComIfGp_getSelectItem(equippedItemIdx);
+    field_0x9BE = itemNo - FLOWER_1;
+    if(isKoukanItem(itemNo) && dComIfGs_getEventReg(l_save_dat[mNpcNo].field_0x02) < 3 && !isGetMap(itemNo)) {
+        if(dComIfGp_event_getTalkXYBtn() == 1) {
+            field_0x9C5 = 0;
+        }
+        else if(dComIfGp_event_getTalkXYBtn() == 2) {
+            field_0x9C5 = 1;
+        }
+        else if(dComIfGp_event_getTalkXYBtn() == 3) {
+            field_0x9C5 = 2;
+        }
+        else {
+            field_0x9C5 = 3;
+        }
+
+        eventIdx = field_0x99E;
+        field_0x9B2 |= 0xC000;
+    }
+    else {
+        if(isGetMap(itemNo)) {
+            eventIdx = field_0x9A4;
+            field_0x9B2 |= 0xC000;
+        }
+        else {
+            eventIdx = dComIfGp_evmng_getEventIdx("DEFAULT_TALK_XY");
+        }
+    }
+
+    return eventIdx;
 }
 
 /* 00003C7C-00003DD4       .text setCollisionB__12daNpcRoten_cFv */
 void daNpcRoten_c::setCollisionB() {
-    /* Nonmatching */
+    MtxP pMtx = mpMorf->getModel()->getAnmMtx(m_bag_jnt_num);
+
+    cXyz temp;
+    temp.x = pMtx[0][3];
+    temp.y = pMtx[1][3];
+    temp.z = pMtx[2][3];
```

Comment:
```suggestion
    cXyz temp;
    mDoMtx_multVecZero(pMtx, &temp);
```
they usually use multveczero to extract the translation of matrix into a cxyz